tedana/metrics/kundu_fit.py

```diff
@@ -240,40 +246,41 @@ def dependence_metrics(catd, tsoc, mmix, t2s, tes, ref_img,
             # Cluster-extent threshold and binarize F-maps
             ccimg = io.new_nii_like(
                 ref_img,
-                np.squeeze(utils.unmask(F_R2_maps[:, i_comp], t2s != 0)))
+                np.squeeze(utils.unmask(F_R2_maps[:, i_comp], mask)))
             F_R2_clmaps[:, i_comp] = utils.threshold_map(
                 ccimg, min_cluster_size=csize, threshold=fmin, mask=mask,
-                binarize=True)
+                binarize=True).astype(bool)
```

Comment:
Why do we need to explicitly cast to boolean after `binarize=True` ? Should that casting instead be performed inside `utils.threshold_map` ?

Reply:
I'm not sure. It's true that binarized data should be boolean, but nibabel is super-mean and for some reason doesn't work with boolean arrays, so there's an argument to be made for using integers as the default for binarized arrays in `threshold_map`.

Reply:
Looking through I think the function is only used in these five instances you're updating, so I think we should just update the function, probably ! If we use it in other ways, we can re-consider the default. WDYT of that idea ?

Reply:
Sounds good. Should be good now.